### ISSUE TITLE
Thread safety analysis - const ref to member not marked as a read

### ISSUE THREAD

**Issue by t-gehring** [llvm/llvm-project#55149]
When compiling with -Wthread-safety to enable the thread safety analysis from https://clang.llvm.org/docs/ThreadSafetyAnalysis.html and for convenience using the macros from the header file there:

```cpp
struct MyStruct {
    int x;
};

Mutex m;
MyStruct data;

// when not holding the lock, correct behaviour for mutable reference
{
    int & mutableRef = data.x; // fine, no write performed
}
{
    int & mutableRef = data.x; // "warning: writing variable 'data' requires holding mutex 'm' exclusively [-Wthread-safety-analysis]"
    mutableRef++; // due to this line performing a write
}
{
    int & mutableRef = data.x; // "warning: reading variable 'data' requires holding mutex 'm' [-Wthread-safety-analysis]"
    int copy = mutableRef; // due to this line performing a read
}
// when not holding the lock, incorrect behaviour for immutable reference
{
    int const & immutableRef = data.x; // no warning emitted
    int copy = immutableRef; // despite this line performing a read
}
```

I believe the last case should give a warning of "warning: reading variable 'data' requires holding mutex 'm' [-Wthread-safety-analysis]".
